pkg/pod/pod.go

```diff
@@ -168,6 +171,15 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 		}
 	}
 
+	// Add env var if hermetic execution was requested
+	if taskRun.Spec.ExecutionMode.Hermetic {
+		for i, s := range stepContainers {
+			// Add it at the end so it overrides
+			env := append(s.Env, corev1.EnvVar{Name: HermeticEnvVar, Value: "1"})
```

Comment:
The order here is very important - it has to come at the end. A few unit tests would be useful to make sure people can't override it with their own env vars. We might also want to make the variable "TEKTON_HERMETIC" to indicate that it's a reserved word.

Reply:
SGTM